includes/global-functions.php

```diff
@@ -203,3 +240,348 @@ function distributor_get_registered_data() {
 
 	return $distributor_registered_data;
 }
+
+/**
+ * Pre-distribute callback for media data.
+ * This is the default pre-distribute callback for media data, used when the type is set to 'media' in distributor_register_data().
+ *
+ * @param mixed $media_id The media ID to be processed before distribution.
+ * @return array The extra data of the media to be distributed to the target site.
+ */
+function distributor_media_pre_distribute_callback( $media_id ) {
+	if ( ! $media_id ) {
+		return array();
+	}
+
+	// Get the media data.
+	$media = get_post( $media_id );
+	if ( ! $media ) {
```

Comment:
A check of the post type would be good here too.

Reply:
Handled in https://github.com/10up/distributor/pull/1309/commits/2c6994ec7ed58c064d282cc2929f128b2b33a428